tee-worker/litentry/core/assertion-build/src/a10.rs

```diff
@@ -60,42 +60,44 @@ pub fn build(
 	let mut client = GraphQLClient::new();
 	let mut found = false;
 	let mut from_date_index = 0_usize;
+	let mut addresses = vec![];
 
 	for id in identities {
-		if found {
-			break
-		}
-
 		if let Identity::Evm { network, address } = id {
 			if matches!(network, EvmNetwork::Ethereum) {
 				let mut address = account_id_to_string(address.as_ref());
 				address.insert_str(0, "0x");
-				debug!("	[AssertionBuild] A10 Ethereum address : {}", address);
+				debug!("Assertion A10 Ethereum address : {}", address);
 
-				let addresses = vec![address.to_string()];
-				for (index, from_date) in ASSERTION_FROM_DATE.iter().enumerate() {
-					// if found is true, no need to check it continually
-					if found {
-						from_date_index = index + 1;
-						break
-					}
+				addresses.push(address);
+			}
+		}
+	}
+
+	if !addresses.is_empty() {
+		for (index, from_date) in ASSERTION_FROM_DATE.iter().enumerate() {
+			if found {
+				from_date_index = index + 1;
```

Comment:
Is this logic correct?

it seems `from_date_index` can only be in the range [2, 6] (i.e. it can never go below 2 as this branch will only be entered when `index` is >= 1)

Apart from that, imagine the user holds since `2023-01-01`, `is_hodler` will only be true when `index == 6`, but the corresponding `from_date_index` is still 0 (as the whole loop ends already).

I don't quite understand why we need to `+1` here